steam/core/connection.py

```diff
@@ -28,6 +28,10 @@ def __init__(self):
 
         self.event_connected = event.Event()
 
+    @property
+    def local_address(self):
+        return self.socket.getsockname()[0] if self.event_connected.is_set() else None
```

Comment:
Lose the condition. `getsockname` will return `0.0.0.0` if the socket is not connected.